pixeltable/catalog/catalog.py

```diff
@@ -423,11 +403,49 @@ def begin_xact(
                         tv.is_validated = False
 
                 if has_exc:
-                    # purge all modified TableVersion instances, we can't guarantee they are still consistent with the
+                    # purge all modified TableVersion instances; we can't guarantee they are still consistent with the
                     # stored metadata
                     for handle in self._modified_tvs:
                         self._clear_tv_cache(handle.id, handle.effective_version)
-                self._modified_tvs = set()
+                self._modified_tvs.clear()
+
+    def raise_from_sql_exc(
```

Comment:
convert_sql_exc()? I had to look this up a few times to remind myself that it's raising a pxt.Error.